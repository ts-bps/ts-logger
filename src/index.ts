import chalk from "chalk";
import * as is from "is";

export const getLogger = ({ log } = { log: console.log }) => {
  return {
    info: (message: any) =>
      log(`
ℹ️ : ${chalk.yellow(
        is.object(message) ? JSON.stringify(message, null, 2) : message
      )}
`),
    success: (message: any) =>
      log(`
✅ : ${chalk.green.bold(
        is.object(message) ? JSON.stringify(message, null, 2) : message
      )}
`),
    error: (message: any) =>
      log(`
🔥 : ${chalk.red.bold(
        is.object(message) ? JSON.stringify(message, null, 2) : message
      )}
    `)
  };
};
