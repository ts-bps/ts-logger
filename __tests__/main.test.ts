import { getLogger } from "../src/main";

describe("getLogger", () => {
  test("exports", () => {
    expect(getLogger()).toMatchSnapshot();
  });
  test("works with strings", () => {
    const mockLog = jest.fn();
    const log = getLogger({ log: mockLog });
    log.success("success");
    log.info("info");
    log.error("info");
    expect(mockLog.mock.calls).toMatchSnapshot();
  });
  test("works with objects", () => {
    const mockLog = jest.fn();
    const log = getLogger({ log: mockLog });
    log.success({ t: "success" });
    log.info({ t: "info" });
    log.error({ t: "info" });
    expect(mockLog.mock.calls).toMatchSnapshot();
  });
});
