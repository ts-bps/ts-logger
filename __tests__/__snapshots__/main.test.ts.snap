// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLogger exports 1`] = `
Object {
  "error": [Function],
  "info": [Function],
  "success": [Function],
}
`;

exports[`getLogger works with objects 1`] = `
Array [
  Array [
    "
‚úÖ : [32m[1m{[22m[39m
[32m[1m  \\"t\\": \\"success\\"[22m[39m
[32m[1m}[22m[39m
",
  ],
  Array [
    "
‚ÑπÔ∏è : [33m{[39m
[33m  \\"t\\": \\"info\\"[39m
[33m}[39m
",
  ],
  Array [
    "
üî• : [31m[1m{[22m[39m
[31m[1m  \\"t\\": \\"info\\"[22m[39m
[31m[1m}[22m[39m
    ",
  ],
]
`;

exports[`getLogger works with strings 1`] = `
Array [
  Array [
    "
‚úÖ : [32m[1msuccess[22m[39m
",
  ],
  Array [
    "
‚ÑπÔ∏è : [33minfo[39m
",
  ],
  Array [
    "
üî• : [31m[1minfo[22m[39m
    ",
  ],
]
`;
